tests/migrations/test_optimizer.py

```diff
@@ -696,3 +700,91 @@ def test_optimize_elidable_operation(self):
         )
+
+    def test_default_create_model_add_field(self):
+        """
+        #26223. AddField optimizing into CreateModel should drop
+        default if preserve_default is False.
+        """
+        self.assertOptimizesTo(
+            [
+                migrations.CreateModel("Foo", [("name", models.CharField(max_length=255))]),
+                migrations.AddField("Foo", "value", models.IntegerField(default=42), preserve_default=False),
+            ],
+            [
+                migrations.CreateModel("Foo", [
+                    ("name", models.CharField(max_length=255)),
+                    ("value", models.IntegerField()),
+                ]),
+            ],
+        )
+
+    def test_default_create_model_alter_field(self):
+        """
+        #26223. AlterField optimizing into CreateModel should drop
+        default if preserve_default is False.
+        """
+        self.assertOptimizesTo(
+            [
+                migrations.CreateModel("Foo", [("value", models.IntegerField(null=True))]),
+                migrations.AlterField("Foo", "value", models.IntegerField(default=42), preserve_default=False),
+            ],
+            [
+                migrations.CreateModel("Foo", [("value", models.IntegerField())]),
+            ],
+        )
+
+    def test_default_add_field_alter_field(self):
+        """
+        #26223. AddField and AlterField optimizing when one of them has
+        preserve_default=False should pass the preserve_default value.
+        """
+        self.assertOptimizesTo(
+            [
+                migrations.AddField("Foo", "value", models.IntegerField(null=True)),
+                migrations.AlterField("Foo", "value", models.IntegerField(default=42), preserve_default=False),
+            ],
+            [
+                migrations.AddField("Foo", "value", models.IntegerField(default=42), preserve_default=False),
+            ],
+        )
+        self.assertOptimizesTo(
+            [
+                migrations.AddField("Foo", "value", models.IntegerField(default=42), preserve_default=False),
+                migrations.AlterField("Foo", "value", models.IntegerField("Value")),
+            ],
+            [
+                migrations.AddField("Foo", "value", models.IntegerField("Value", default=42), preserve_default=False),
+            ],
+        )
+
+    def test_default_add_field_rename_field(self):
+        """
+        #26223. AddField optimizing with RenameField should retain its
+        preserve_default value.
+        """
+        self.assertOptimizesTo(
+            [
+                migrations.AddField("Foo", "value", models.IntegerField(default=42), preserve_default=False),
+                migrations.RenameField("Foo", "value", "price"),
+            ],
+            [
+                migrations.AddField("Foo", "price", models.IntegerField(default=42), preserve_default=False),
+            ],
+        )
+
+    def test_default_alter_field_rename_field(self):
```

Comment:
The optimization was already here before. I think the idea is to move `RenameField` upwards where it could possibly merge into an `AddField` or `CreateModel`.

What's relevant to this ticket is that previously after the swapping `AlterField` was missing its `preserve_default=False`.